src/smartstream/examples/map_regex/src/lib.rs

```diff
@@ -0,0 +1,20 @@
+use fluvio_smartstream::{smartstream, Record, RecordData};
+use regex::Regex;
+
+#[smartstream(map)]
+pub fn map(record: &Record) -> (Option<RecordData>, RecordData) {
+    let key = record.key.clone();
+
+    let string_result = std::str::from_utf8(record.value.as_ref());
+    let string = match string_result {
+        Ok(s) => s,
+        Err(_) => return (key, record.value.clone()),
+    };
+
+    let social_security_regex = Regex::new(r"\d{3}-\d{2}-\d{4}").unwrap();
```

Comment:
this should be declared as constant or outside function.  Creating regex every time in the loop is costly

Reply:
I thought about that, I was originally going to put this into a `once_cell::sync::Lazy`, but then I didn't know whether the Wasm module was re-instantiated every time a record was passed or not. If we put this into a `static Lazy<Regex>` will it be preserved between executions? I think it will at least for a batch at a time now that I think about it. I'll go ahead and add that.